app/tests/data_factory.py

```diff
@@ -0,0 +1,978 @@
+            name: string
+                Overwrite the sample name
+            **kwargs:
+                any parameter will overwrite existing attributes
+        '''
+        self.attr_name = name
+        super().__init__(sample=sample, item=item, name=name, **kwargs)
+
+        file = getattr(self, 'attr_file', None)
+        if file:
+            self.attr_checksum_multihash = get_sha256_multihash(file)
+            self.attr_file = SimpleUploadedFile(
+                f'{item.collection.name}/{item.name}/{self.attr_name}', file
+            )
+
+    @property
+    def json(self):
+        '''Returns a json serializable representation of the sample data
+
+        This json payload can then be used in the write API payload or to check
+        the read API payload.
+
+        Returns
+            A dictionary with the sample data
+        '''
+        # pylint: disable=no-member
+        data = super().json
+        item = data.pop('item')
+        data['item'] = item.name
+        file = data.pop('file', None)
+        if file:
+            data['href'] = \
+                f'http://{settings.AWS_S3_CUSTOM_DOMAIN}/{item.collection.name}/{item.name}/{file.name}'
+        return data
+
+
+class FactoryBase:
+    '''Factory base class
+    '''
+    factory_name = 'base'
+    sample_class = SampleData
+
+    def __init__(self):
+        self.last = None
+
+    @classmethod
+    def get_last_name(cls, last):
+        '''Return a factory name incremented by one (e.g. 'collection-1')
+        '''
+        if last is None:
+            last = f'{cls.factory_name}-0'
+        last = '{}-{}'.format(
+            cls.factory_name, int(re.match(fr"{cls.factory_name}-(\d+)", last).group(1)) + 1
+        )
+        return last
+
+    def create_sample(self, name=None, db_create=False, **kwargs):
+        '''Create a data sample
+
+        Args:
+            name: string
+                Data name, if not given it creates a '{factory_name}-n' with n being incremented
+                after each function call
+            sample: string
+                sample based on the sample named found in tests.sample_data
+            **kwargs:
+                key/value pairs used to overwrite arbitrary attribute in the sample
+
+        Returns:
+            The data sample
+        '''
+        if name:
+            data_sample = self.sample_class(name=name, **kwargs)
+        else:
+            self.last = self.get_last_name(self.last)
+            data_sample = self.sample_class(name=self.last, **kwargs)
+        if db_create:
+            data_sample.create()
+        return data_sample
+
+    def create_samples(self, samples, db_create=False, kwargs_list=True, **kwargs):
+        '''Creates several samples
+
+        Args:
+            samples: integer | [string]
+                number of DB sample to create or list of sample data name
+            db_create: bool
+                create the sample in the DB
+            sample: string
```

Comment:
is this an obsolete comment, as `sample` seems to be no longer an argument of `create_samples` any more